src/Router/BaseRouter.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Implements BaseRouter, a Router to handle the base of a LORIS
+ * install.
+ *
+ * PHP Version 7
+ *
+ * @category Router
+ * @package  Router
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\Router;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+use \LORIS\Http\StringStream;
+
+/**
+ * Handles the root of a LORIS install. It will mostly delegate to the
+ * A BaseRouter handles the base of a LORIS install and delegates to
+ * other routers (such as Module routers)
+ *
+ * @category Router
+ * @package  Router
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class BaseRouter extends PrefixRouter implements RequestHandlerInterface
+{
+    protected $projectdir;
+    protected $moduledir;
+    protected $user;
+
+    /**
+     * Construct a BaseRouter
+     *
+     * @param \User  $user       The user accessing LORIS. (May be an AnonymousUser
+     *                           instance).
+     * @param string $projectdir The base of the LORIS project directory.
+     * @param string $moduledir  The base of the LORIS modules directory.
+     */
+    public function __construct(\User $user, string $projectdir, string $moduledir)
+    {
+        $this->user       = $user;
+        $this->projectdir = $projectdir;
+        $this->moduledir  = $moduledir;
+    }
+
+    /**
+     * Handle delegates to an appropriate sub-router to do the real handling of a
+     * LORIS request.
+     *
+     * @param ServerRequestInterface $request The PSR7 request
+     *
+     * @return ServerResponse The PSR15 response.
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $uri  = $request->getUri();
+        $path = $uri->getPath();
+        // Replace multiple slashes in the URL with a single slash
+        $path = preg_replace("/\/+/", "/", $path);
+        // Remove a training slash remaining, so that foo/ and foo are the same
```

Comment:
Is this properly written? there are lots of ...ing words.

Reply:
it should be 'any trailing'.. but I think that's the only problem?

Reply:
Yes, that would be better

Reply:
Fixed